--- a/tcplay_api.c
+++ b/tcplay_api.c
@@ -114,7 +114,8 @@ tc_api_create_volume(tc_api_opts *api_opts)
 	    check_prf_algo(api_opts->tc_prf_hash_hidden, 1),
 	    check_cipher_chain(api_opts->tc_cipher_hidden, 1),
 	    api_opts->tc_passphrase, api_opts->tc_passphrase_hidden,
-	    api_opts->tc_size_hidden_in_bytes, 0 /* non-interactive */);
+	    api_opts->tc_size_hidden_in_bytes, 0 /* non-interactive */,
+	    0 /* non-secure erase */);
 
 	return (err) ? TC_ERR : TC_OK;
 }
