--- gtimeutils-9999/Makefile
+++ gtimeutils-9999/Makefile
@@ -1,22 +1,21 @@
-jCC      = gcc
-CFLAGS   = -std=c99 -Wall -Wextra
-LIBS_TIMER = `pkg-config --cflags --libs gtk+-3.0 libnotify libcanberra`
-LIBS_STOPW = `pkg-config --cflags --libs gtk+-3.0`
+CC      ?= gcc
+CFLAGS  += -std=c99 -Wall -Wextra
+LIBS_TIMER = $(shell pkg-config --cflags --libs gtk+-3.0 libnotify libcanberra)
+LIBS_STOPW = $(shell pkg-config --cflags --libs gtk+-3.0)
  
 PREFIX   ?= /usr/local
 BINPREFIX = $(PREFIX)/bin
 
-all: CFLAGS += -Os
 all: gstopwatch gtimer
 
-debug: CFLAGS += -O0 -g -pedantic -DDEBUG
+debug: CFLAGS += -pedantic -DDEBUG
 debug: all
 
 gstopwatch: gstopwatch.c
-	$(CC) gstopwatch.c -o gstopwatch $(CFLAGS) $(LIBS_STOPW)
+	$(CC) gstopwatch.c -o gstopwatch $(CFLAGS) $(LIBS_STOPW) ${LDFLAGS}
 
 gtimer: gtimer.c
-	$(CC) gtimer.c -o gtimer $(CFLAGS) $(LIBS_TIMER)
+	$(CC) gtimer.c -o gtimer $(CFLAGS) $(LIBS_TIMER) ${LDFLAGS}
 
 install:
 	mkdir -p $(DESTDIR)/$(BINPREFIX)
