From https://aur.archlinux.org/packages.php?ID=63769

--- cdc-acm.c
+++ cdc-acm.c
@@ -532,11 +532,6 @@
 		goto error_submit_urb;
 	}
 
-	acm->ctrlout = ACM_CTRL_DTR | ACM_CTRL_RTS;
-	if (acm_set_control(acm, acm->ctrlout) < 0 &&
-	    (acm->ctrl_caps & USB_CDC_CAP_LINE))
-		goto error_set_control;
-
 	usb_autopm_put_interface(acm->control);
 
 	/*
@@ -559,8 +554,6 @@
 		usb_kill_urb(acm->read_urbs[i]);
 	acm->ctrlout = 0;
 	acm_set_control(acm, acm->ctrlout);
-error_set_control:
-	usb_kill_urb(acm->ctrlurb);
 error_submit_urb:
 	usb_autopm_put_interface(acm->control);
 error_get_interface:
@@ -594,7 +587,6 @@
 	mutex_lock(&acm->mutex);
 	if (!acm->disconnected) {
 		pm_err = usb_autopm_get_interface(acm->control);
-		acm_set_control(acm, acm->ctrlout = 0);
 
 		for (;;) {
 			urb = usb_get_from_anchor(&acm->delayed);
