--- Eclipse_6.1_187/RUNME
+++ Eclipse_6.1_187/RUNME
@@ -124,7 +124,7 @@
 }
 
 
-TCL_REQUIRED="8.5 8.4 8.3"
+TCL_REQUIRED="8.6 8.5 8.4 8.3"
 
 usable_tcl() {
     [	-x "$TCL_WISH" -a \
@@ -227,10 +227,10 @@
 
     case $ARCH in
 	*_macosx)
-	    ask "What is the version number of your Tcl/Tk installation (8.X) \n[Note that 8.5 prior to 8.5.7 was broken for ECLiPSe]"  TCL_VERSION "8.3 8.4 8.5" 
+	    ask "What is the version number of your Tcl/Tk installation (8.X) \n[Note that 8.5 prior to 8.5.7 was broken for ECLiPSe]"  TCL_VERSION "8.3 8.4 8.5 8.6" 
 	    ;;
 	*)
-	    ask "What is the version number of your Tcl/Tk installation (8.X)" TCL_VERSION "8.3 8.4 8.5"
+	    ask "What is the version number of your Tcl/Tk installation (8.X)" TCL_VERSION "8.3 8.4 8.5 8.6"
 	    ;;
     esac
 
--- Eclipse_6.1_187/configure.ac
+++ Eclipse_6.1_187/configure.ac
@@ -1296,7 +1296,7 @@
 
 if test "$tcl_requested" = "yes"; then
 
-    TCL_REQUIRED="8.5 8.4 8.3"
+    TCL_REQUIRED="8.6 8.5 8.4 8.3"
 
     if test "$OSCLASS" != "win" ; then
 	dnl find X includes and libraries
--- Eclipse_6.1_187/Kernel/src/Makefile.in
+++ Eclipse_6.1_187/Kernel/src/Makefile.in
@@ -94,7 +94,7 @@
 OPT_FLAGS_ARITH = @OPT_FLAGS_ARITH@ 
 LOCAL_FLAGS = @LOCAL_FLAGS@
 FLOAT_ROUND_FLAGS = @FLOAT_ROUND_FLAGS@
-CFLAGS = @CFLAGS@
+CFLAGS = -DUSE_INTERP_RESULT @CFLAGS@
 OBJECTS_SUFFIX = @OBJECTS_SUFFIX@
 LIB_PREFIX = @LIB_PREFIX@
 
