--- Eclipse_6.1_187/configure.ac
+++ Eclipse_6.1_187/configure.ac
@@ -1962,13 +1962,20 @@
 AC_CHECK_PROG(PROG_PS2PDF,ps2pdf,yes,no)
 AC_CHECK_PROG(PROG_HEVEA, hevea, yes,no)
 
+AC_ARG_WITH([doc], [[  --without-doc
+                        Build ECLiPSe without documentation
+    ]],
+    [doc_requested="$withval"], [doc_requested=yes])
+
 MAKE_DOCUMENTS=
+if test "$doc_requested" != "no"; then
 if test "$PROG_LATEX" = yes -a "$PROG_PS2PDF" = yes -a "$PROG_HEVEA" = yes; then
     MAKE_DOCUMENTS="make_documents"
     AC_MSG_NOTICE([Can build ECLiPSe documentation])
 else
     AC_MSG_WARN([No tools to build ECLiPSe documentation])
 fi
+fi
 
 
 #----------------------------------------------------------------------
