--- Eclipse_6.1_187/configure.ac
+++ Eclipse_6.1_187/configure.ac
@@ -1494,7 +1494,7 @@
 
 AC_ARG_WITH([osi], [[  --with-osi[=versions]
 			Build ECLiPSe interface to COIN-OR Osi.  You can
-			optionally specify a version list, e.g. "clpcbc symclp".
+			optionally specify a version list, e.g. "clpcbc symclp glpk".
 			Default is to build all available versions.
     ]],
     [osi_versions_requested="$withval"], [osi_versions_requested=yes])
@@ -1504,34 +1504,22 @@
 OSICLP_ADDDEF=""
 if test "$osi_versions_requested" != "no"; then
     if test "$osi_versions_requested" = "yes"; then
-    	osi_versions_requested="clpcbc symclp"
+    	osi_versions_requested="clpcbc symclp glpk"
     fi
+    OSI_VERSIONS=
+    PKG_CHECK_MODULES(OSI, [osi])
     for version in $osi_versions_requested; do
-	found_osi_version=
-	dnl For backward compatibility, allow ${version}amd as well
-	OSIDIRS="$OSIBASEDIR${version}amd $OSIBASEDIR$version"
-	for OSIDIR in $OSIDIRS; do
-	    if test -d "$OSIDIR/$ARCH/lib"; then
-		found_osi_version=yes
-		OSI_VERSIONS="$OSI_VERSIONS $version"
-		AC_MSG_NOTICE([Using OSI $version in $OSIDIR])
-		case $version in
-		clpcbc) CLPCBCDIR="$OSIDIR" ;;
-		symclp) SYMCLPDIR="$OSIDIR" ;;
-		glpk)  OSIGLPKDIR="$OSIDIR" ;;
-		esac
-		ls -d $OSIDIR/$ARCH/lib/*amd* >/dev/null 2>/dev/null
-		if test $? -eq 0; then
-		     AC_MSG_NOTICE([Found AMD support for $version])
-		     OSICLP_ADDLIB="-lamd"
-		     OSICLP_ADDDEF="-DUFL_BARRIER"
-		fi
-		break
-	    fi
-	done
-	if test -z "$found_osi_version"; then
-	    AC_MSG_WARN([Could not find $ARCH support for $version in $OSIDIRS])
-	fi
+	case $version in
+            clpcbc)
+                PKG_CHECK_MODULES(CLPCBC, [osi-cbc], [CLPCBCDIR="yes" ; OSI_VERSIONS="$OSI_VERSIONS $version"])
+		 ;;
+            symclp)
+                PKG_CHECK_MODULES(SYMCLP, [osi-sym], [SYMCLPDIR="yes" ; OSI_VERSIONS="$OSI_VERSIONS $version"])
+		 ;;
+            glpk)
+                PKG_CHECK_MODULES(OSIGLPK, [osi-glpk], [OSIGLPKDIR="yes" ; OSI_VERSIONS="$OSI_VERSIONS $version"])
+		 ;;
+        esac
     done
 fi
 if test -n "$OSI_VERSIONS"; then
--- Eclipse_6.1_187/Eplex/Makefile.in
+++ Eclipse_6.1_187/Eplex/Makefile.in
@@ -96,34 +96,34 @@
 SYMCLPDIR = @SYMCLPDIR@
 OSIGLPKDIR = @OSIGLPKDIR@
 
-CLPCBCARCHDIR = $(CLPCBCDIR)/$(ARCH)
-SYMCLPARCHDIR = $(SYMCLPDIR)/$(ARCH)
-OSIGLPKARCHDIR = $(OSIGLPKDIR)/$(ARCH)
-
-COINOSILIBS = -lOsi -lCoinUtils
-COINCLPLIBS = -lOsiClp -lCgl -lClp 
-COINCBCLIBS = -lCbcSolver -lCbc
+CLPCBC_CFLAGS = @CLPCBC_CFLAGS@
+SYMCLP_CFLAGS = @SYMCLP_CFLAGS@
+OSIGLPK_CFLAGS = @OSIGLPK_CFLAGS@
+
+CLPCBC_LIBS = @CLPCBC_LIBS@
+SYMCLP_LIBS = @SYMCLP_LIBS@
+OSIGLPK_LIBS = @OSIGLPK_LIBS@
 
 $(ARCH)/seosiclpcbc.$(OBJ_SUFFIX): eplex.c coinplex.cpp eplex_coin.h eplex_coin.c eplex.h coinplex_params.h 
-	$(CXX) -DCOIN -DCOIN_USE_CLP @OSICLP_ADDDEF@ $(CPPFLAGS) $(CFLAGS) -I$(CLPCBCARCHDIR)/include -c coinplex.cpp
+	$(CXX) -DCOIN -DCOIN_USE_CLP @OSICLP_ADDDEF@ $(CPPFLAGS) $(CFLAGS) $(CLPCBC_CFLAGS) -c coinplex.cpp
 	$(CC) -DCOIN -DCOIN_USE_CLP -DC_TO_COIN $(CPPFLAGS) $(CFLAGS) -c eplex.c
-	$(DYLD) $(DYNLDFLAGS) eplex.o coinplex.o $(CLPCBCARCHDIR)/lib/CbcBranchUser.o $(CLPCBCARCHDIR)/lib/CbcCompareUser.o -L$(CLPCBCARCHDIR)/lib  $(COINCBCLIBS) $(COINCLPLIBS) @OSICLP_ADDLIB@ $(COINOSILIBS) $(STDCXXLIB) $(ECLIBS) $(GMP_LD) -o seosiclpcbc.$(OBJ_SUFFIX)
+	$(DYLD) $(DYNLDFLAGS) eplex.o coinplex.o $(CLPCBC_LIBS) @OSICLP_ADDLIB@ $(STDCXXLIB) $(ECLIBS) $(GMP_LD) -o seosiclpcbc.$(OBJ_SUFFIX)
 	[ -d $(ARCH) ] || mkdir $(ARCH)
-	mv seosiclpcbc.$(OBJ_SUFFIX) $(ARCH)/seosiclpcbc.$(OBJ_SUFFIX)
+	cp seosiclpcbc.$(OBJ_SUFFIX) $(ARCH)/seosiclpcbc.$(OBJ_SUFFIX)
 
 $(ARCH)/seosisymclp.$(OBJ_SUFFIX): eplex.c coinplex.cpp eplex_coin.h eplex_coin.c eplex.h 
-	$(CXX) -DCOIN -DCOIN_USE_SYM $(CPPFLAGS) $(CFLAGS) -I$(SYMCLPARCHDIR)/include -c coinplex.cpp
+	$(CXX) -DCOIN -DCOIN_USE_SYM $(CPPFLAGS) $(CFLAGS) $(SYMCLP_CFLAGS) -c coinplex.cpp
 	$(CC) -DCOIN -DC_TO_COIN $(CPPFLAGS) $(CFLAGS) -c eplex.c
-	$(DYLD) $(DYNLDFLAGS) eplex.o coinplex.o -L$(SYMCLPARCHDIR)/lib -lOsiSym -lSym $(COINCLPLIBS) $(COINOSILIBS) $(STDCXXLIB) $(ECLIBS) $(GMP_LD) -o seosisymclp.$(OBJ_SUFFIX)
+	$(DYLD) $(DYNLDFLAGS) eplex.o coinplex.o $(SYMCLP_LIBS) $(STDCXXLIB) $(ECLIBS) $(GMP_LD) -o seosisymclp.$(OBJ_SUFFIX)
 	[ -d $(ARCH) ] || mkdir $(ARCH)
-	mv seosisymclp.$(OBJ_SUFFIX) $(ARCH)/seosisymclp.$(OBJ_SUFFIX)
+	cp seosisymclp.$(OBJ_SUFFIX) $(ARCH)/seosisymclp.$(OBJ_SUFFIX)
 
 $(ARCH)/seosiglpk.$(OBJ_SUFFIX): eplex.c coinplex.cpp eplex_coin.h eplex_coin.c eplex.h 
-	$(CXX) -DCOIN -DCOIN_USE_GLPK $(CPPFLAGS) $(CFLAGS) -I$(OSIGLPKARCHDIR)/include -c coinplex.cpp
+	$(CXX) -DCOIN -DCOIN_USE_GLPK $(CPPFLAGS) $(CFLAGS) $(OSIGLPK_CFLAGS) -c coinplex.cpp
 	$(CC) -DCOIN -DC_TO_COIN $(CPPFLAGS) $(CFLAGS) -c eplex.c
-	$(DYLD) $(DYNLDFLAGS) eplex.o coinplex.o -L$(OSIGLPKARCHDIR)/lib -lOsiGlpk -lglpk  $(COINOSILIBS) $(STDCXXLIB) $(ECLIBS) $(GMP_LD) -o seosiglpk.$(OBJ_SUFFIX)
+	$(DYLD) $(DYNLDFLAGS) eplex.o coinplex.o $(OSIGLPK_LIBS) $(STDCXXLIB) $(ECLIBS) $(GMP_LD) -o seosiglpk.$(OBJ_SUFFIX)
 	[ -d $(ARCH) ] || mkdir $(ARCH)
-	mv seosiglpk.$(OBJ_SUFFIX) $(ARCH)/seosiglpk.$(OBJ_SUFFIX)
+	cp seosiglpk.$(OBJ_SUFFIX) $(ARCH)/seosiglpk.$(OBJ_SUFFIX)
 
 #----------------------------------------------------------------------
 # EPLEX/CPLEX
