--- Eclipse_6.1_187/icparc_solvers/Makefile.in
+++ Eclipse_6.1_187/icparc_solvers/Makefile.in
@@ -47,7 +47,7 @@
 GMP_LD = @GMP_LD@
 FLOAT_ROUND_FLAGS = @FLOAT_ROUND_FLAGS@
 
-CPPFLAGS  = -I. -I$(ECLIPSEDIR)/include/$(ARCH) -I$(ECLIPSEDIR)/Kernel/src -I- $(OS_INCLUDES)
+CPPFLAGS  = -fPIC -I. -I$(ECLIPSEDIR)/include/$(ARCH) -I$(top_srcdir)/Kernel/src -iquote $(OS_INCLUDES)
 STDCXXLIB = @STDCXXLIB@
 ECLIBS	= -L$(ECLIPSEDIR)/lib/$(ARCH) -leclipse @LIBS@
 
@@ -84,14 +84,14 @@
 ifeq ($(OSCLASS),win)
 
 $(ARCH)/edge_finder.dll: edge_finder.c
-	[ -d $(ARCH) ] || mkdir $(ARCH)
+	mkdir -p $(ARCH)
 	$(CC) $(CPPFLAGS) $(CFLAGS) -shared -o $@ \
 		$(<:.c=.def) $< $(ECLIBS)
 
 else
 
 $(ARCH)/edge_finder.$(OBJ_SUFFIX): edge_finder.c
-	[ -d $(ARCH) ] || mkdir $(ARCH)
+	mkdir -p $(ARCH)
 	$(CC) $(CPPFLAGS) $(CFLAGS) $(LOCAL_FLAGS) edge_finder.c \
               -c -o $(ARCH)/edge_finder.o
 	$(DYLD) $(DYNLDFLAGS) $(GMP_LD) $(ECLIBS) \
@@ -101,14 +101,14 @@
 ifeq ($(OSCLASS),win)
 
 $(ARCH)/edge_finder3.dll: edge_finder.c
-	[ -d $(ARCH) ] || mkdir $(ARCH)
+	mkdir -p $(ARCH)
 	$(CC) -DSTRONG $(CPPFLAGS) $(CFLAGS) -shared -o $@ \
 		$(<:.c=.def) $< $(ECLIBS)
 
 else
 
 $(ARCH)/edge_finder3.$(OBJ_SUFFIX): edge_finder3.c
-	[ -d $(ARCH) ] || mkdir $(ARCH)
+	mkdir -p $(ARCH)
 	$(CC) -DSTRONG $(CPPFLAGS) $(CFLAGS) $(LOCAL_FLAGS) edge_finder3.c \
               -c -o $(ARCH)/edge_finder3.o
 	$(DYLD) $(DYNLDFLAGS) $(GMP_LD) $(ECLIBS) \
@@ -123,14 +123,14 @@
 ifeq ($(OSCLASS),win)
 
 $(ARCH)/bitmap.dll: bitmap.c
-	[ -d $(ARCH) ] || mkdir $(ARCH)
+	mkdir -p $(ARCH)
 	$(CC) $(CPPFLAGS) $(CFLAGS) -shared -o $@ \
 		$(<:.c=.def) $< $(ECLIBS)
 
 else
 
 $(ARCH)/bitmap.$(OBJ_SUFFIX): bitmap.c
-	[ -d $(ARCH) ] || mkdir $(ARCH)
+	mkdir -p $(ARCH)
 	$(CC) $(CPPFLAGS) $(CFLAGS) $(LOCAL_FLAGS) bitmap.c \
               -c -o $(ARCH)/bitmap.o
 	$(DYLD) $(DYNLDFLAGS) $(GMP_LD) $(ECLIBS) \
@@ -145,19 +145,19 @@
 ifeq ($(OSCLASS),win)
 
 $(ARCH)/ic.dll: ic.c $(ARCH)/bitmap.dll
-	[ -d $(ARCH) ] || mkdir $(ARCH)
+	mkdir -p $(ARCH)
 	$(CC) $(CPPFLAGS) $(CFLAGS) -shared -o $@ \
 		$(<:.c=.def) $< $(ARCH)/bitmap.dll $(ECLIBS)
 else
 
 $(ARCH)/ic.$(OBJ_SUFFIX): ic.c $(ARCH)/bitmap.$(OBJ_SUFFIX)
-	[ -d $(ARCH) ] || mkdir $(ARCH)
+	mkdir -p $(ARCH)
 	$(CC) $(CPPFLAGS) $(CFLAGS) $(FLOAT_ROUND_FLAGS) \
               ic.c -c -o $(ARCH)/ic.o
 	$(DYLD) $(DYNLDFLAGS)  $(ECLIBS) $(GMP_LD) \
                $(ARCH)/ic.o $(ARCH)/bitmap.$(OBJ_SUFFIX) -o $(ARCH)/ic.$(OBJ_SUFFIX) 
 #i386_linux/ic.so: ic2.c
-#	[ -d i386_linux ] || mkdir i386_linux
+#	mkdir -p i386_linux
 #	$(CC) -g $(CPPFLAGS) $(CFLAGS) -shared ic2.c $(ECLIBS) -o i386_linux/ic.so
 
 endif 
@@ -178,14 +178,14 @@
 ifeq ($(OSCLASS),win)
 
 $(ARCH)/eregex.dll: eregex.c $(ARCH)/libregex.a
-	[ -d $(ARCH) ] || mkdir $(ARCH)
+	mkdir -p $(ARCH)
 	$(CC) $(CPPFLAGS) $(CFLAGS) -Irxspencer -shared -o $@ \
 		$(<:.c=.def) $< $(ARCH)/libregex.a $(ECLIBS)
 
 else
 
 $(ARCH)/eregex.$(OBJ_SUFFIX): eregex.c
-	[ -d $(ARCH) ] || mkdir $(ARCH)
+	mkdir -p $(ARCH)
 	$(CC) $(CPPFLAGS) $(CFLAGS) $(LOCAL_FLAGS) eregex.c \
               -c -o $(ARCH)/eregex.o
 	$(DYLD) $(DYNLDFLAGS) $(GMP_LD) $(ECLIBS) $(RXLIBS) \
@@ -196,7 +196,7 @@
 # CAUTION: we build directly in the rxspencer directory, but the binary
 # libregex.a is moved after building to allow different architectures 
 $(ARCH)/libregex.a:
-	[ -d $(ARCH) ] || mkdir $(ARCH)
+	mkdir -p $(ARCH)
 	( cd rxspencer ; \
 	  CC="$(CC)" AR="$(AR)" $(MAKE) lib)
 	mv rxspencer/libregex.a $(ARCH)/.
