--- net-tools-0.89_p20140724/Makefile
+++ net-tools-0.89_p20140724/Makefile
@@ -33,7 +33,7 @@
 NET_LIB_PATH = lib
 NET_LIB_NAME = net-tools
 
-PROGS	:= ifconfig arp netstat route rarp slattach plipconfig nameif
+PROGS	:= ifconfig arp netstat route rarp slattach plipconfig nameif hostname
 
 -include config.make
 ifeq ($(HAVE_IP_TOOLS),1)
@@ -202,6 +202,7 @@
 	install -m 0755 ifconfig   ${BASEDIR}${SBINDIR}
 	install -m 0755 nameif     ${BASEDIR}${SBINDIR}
 	install -m 0755 netstat    ${BASEDIR}${BINDIR}
+	install -m 0755 hostname   ${BASEDIR}${BINDIR}
 	install -m 0755 plipconfig $(BASEDIR)${SBINDIR}
 	install -m 0755 rarp       ${BASEDIR}${SBINDIR}
 	install -m 0755 route      ${BASEDIR}${SBINDIR}
